src/backend/distributed/transaction/coordinated_transaction_management.c

```diff
@@ -73,6 +74,9 @@ BeginCoordinatedTransaction(void)
 	}
 
 	CurrentCoordinatedTransactionState = COORD_TRANS_STARTED;
+
+	/* this is crucial for managing distributed transaction ids */
+	GenerateAndSetNextDistributedTransactionId();
```

Comment:
That comment doesn't actually add anything...  I'd maybe rename the function to AssignDistributedTransactionId() - the "next" bit is superflous to know about at this level.